File: shared/src/commonMain/kotlin/com/kakapo/oakane/presentation/viewModel/goal/GoalState.kt
package com.kakapo.oakane.presentation.viewModel.goal

data class GoalState(
    val fileName: String = "",
    val goalName: String = "",
    val saved: Double = 0.0,
    val target: Double = 0.0,
    val startDate: Long = 0L,
    val endDate: Long = 0L,
    val note: String = ""
){
    val progress: Float get() {
        return (saved / target).toFloat()
    }

    val dayLeft: String get() {
        return (endDate - startDate).toString()
    }
}

File: iosApp/iosApp/viewModel/goal/GoalState.swift
import Foundation
import Shared

struct GoalState {
    var fileName: String = ""
    var title: String = ""
    var savedAmount: String = ""
    var currentAmount: String = ""
    var targetAmount: String = ""
    var progress: Float = 0.0
    var startDate: Int64 = 0
    var endDate: Int64 = 0
    var daysLeft: Int64 = 0
    var note: String = ""
    var isDialogShown: Bool = false
    var dialogContent: GoalDialogContent = .updateAmount
    
    init(){}
    
    init(state: GoalStateKt){
        fileName = state.goal.fileName
        title = state.goal.name
        targetAmount = state.targetAmount
        progress = state.goal.progress
        startDate = state.goal.startDate
        endDate = state.goal.endDate
        daysLeft = state.dayLeft
        note = state.goal.note
        isDialogShown = state.dialogShown
        dialogContent = state.dialogContent
        currentAmount = state.currentAmount
    }
}
